src/deepsparse/transformers/pipelines.py

```diff
@@ -527,42 +524,47 @@ def __call__(self, inputs: Union[str, List[str]], **kwargs):
 
         answers = []
 
-        for i, sentence in enumerate(_inputs):
+        tokens = self.tokenizer(
+            _inputs,
+            return_tensors=self._framework,
+            truncation=TruncationStrategy.LONGEST_FIRST.value,
+            padding=PaddingStrategy.MAX_LENGTH.value,
+            return_special_tokens_mask=True,
+            return_offsets_mapping=self.tokenizer.is_fast,
+        )
 
-            tokens = self.tokenizer(
-                sentence,
-                return_tensors=self._framework,
-                truncation=TruncationStrategy.LONGEST_FIRST.value,
-                padding=PaddingStrategy.MAX_LENGTH.value,
-                return_special_tokens_mask=True,
-                return_offsets_mapping=self.tokenizer.is_fast,
-            )
-            if self.tokenizer.is_fast:
-                offset_mapping = tokens.pop("offset_mapping")[0]
-            elif offset_mappings:
-                offset_mapping = offset_mappings[i]
-            else:
-                offset_mapping = None
+        if self.tokenizer.is_fast:
+            offset_mapping = tokens.pop("offset_mapping")
+        elif not offset_mappings:
+            offset_mapping = [None] * len(_inputs)
+
+        special_tokens_mask = tokens.pop("special_tokens_mask")
 
-            special_tokens_mask = tokens.pop("special_tokens_mask")[0]
+        # Forward
+        _forward_pass = self._forward(tokens)
+        for entities_index, current_entities in enumerate(_forward_pass[0]):
 
-            # Forward
-            entities = self._forward(tokens)[0][0]
-            input_ids = tokens["input_ids"][0]
+            input_ids = tokens["input_ids"][entities_index]
 
-            scores = np.exp(entities) / np.exp(entities).sum(-1, keepdims=True)
+            scores = np.exp(current_entities) / np.exp(current_entities).sum(
+                -1, keepdims=True
+            )
             pre_entities = self.gather_pre_entities(
-                sentence, input_ids, scores, offset_mapping, special_tokens_mask
+                _inputs[entities_index],
+                input_ids,
+                scores,
+                offset_mapping[entities_index],
+                special_tokens_mask[entities_index],
             )
             grouped_entities = self.aggregate(pre_entities, self.aggregation_strategy)
             # Filter anything that is in self.ignore_labels
-            entities = [
+            current_entities = [
                 entity
                 for entity in grouped_entities
                 if entity.get("entity", None) not in self.ignore_labels
                 and entity.get("entity_group", None) not in self.ignore_labels
             ]
-            answers.append(entities)
+            answers.append(current_entities)
```

Comment:
this code looks similar and maybe identical to `examples/huggingface-transformers/pipelines.py`, could this be shared? especially since pipelines is in the deepsparse package

Reply:
Good catch, thanks @mgoin This will be removed in a separate diff, docs and usages would need to be updated.

<img src="https://media0.giphy.com/media/AMQgg6x2QLCFoG1Whr/giphy.gif"/>